providers/configcat/src/main/java/dev/openfeature/contrib/providers/configcat/ConfigCatProvider.java

```diff
@@ -0,0 +1,176 @@
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+        return getEvaluation(Boolean.class, key, defaultValue, ctx);
+    }
+
+    @Override
+    public ProviderEvaluation<String> getStringEvaluation(String key, String defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+        return getEvaluation(String.class, key, defaultValue, ctx);
+    }
+
+    @Override
+    public ProviderEvaluation<Integer> getIntegerEvaluation(String key, Integer defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+        return getEvaluation(Integer.class, key, defaultValue, ctx);
+    }
+
+    @Override
+    public ProviderEvaluation<Double> getDoubleEvaluation(String key, Double defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+        return getEvaluation(Double.class, key, defaultValue, ctx);
+    }
+
+    @SneakyThrows
+    @Override
+    public ProviderEvaluation<Value> getObjectEvaluation(String key, Value defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+        User user = ctx == null ? null : dev.openfeature.contrib.providers.configcat.ContextTransformer.transform(ctx);
+        EvaluationDetails<String> evaluationDetails = configCatClient
+            .getValueDetails(String.class, key, user, defaultValue.asString());
+        return ProviderEvaluation.<Value>builder()
+            .value(Value.objectToValue(evaluationDetails.getValue()))
+            .variant(evaluationDetails.getVariationId())
+            .errorMessage(evaluationDetails.getError())
+            .build();
+    }
+
+    private <T> ProviderEvaluation<T> getEvaluation(Class<T> classOfT, String key, T defaultValue,
+           EvaluationContext ctx) {
+        User user = ctx == null ? null : ContextTransformer.transform(ctx);
```

Comment:
I think the null check can be moved to the `ContextTransformer.transform` method.

Reply:
not always expected to pass null to a single parameter method, I prefer to leave it like this to be consistent with other providers.